### PR TITLE
Add test for setForceHighestSupportedBitrate in DefaultTrackSelector

### PR DESCRIPTION
**Pull request by BetweenTwoBits**
There is already a test for setForceLowestBitrate(boolean forceLowestBitrate). This PR adds a test for the opposing method setForceHighestSupportedBitrate(boolean forceHighestSupportedBitrate).

### CODE REVIEW

File: library/core/src/test/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelectorTest.java [hunk 1]
```diff
@@ -1078,6 +1078,33 @@ public void testSelectTracksWithinCapabilitiesAndForceLowestBitrateSelectLowerBi
...
+                    Format.NO_VALUE, 2, 44100, null, null, 0, null);
+
+    TrackSelectorResult result =
+            trackSelector.selectTracks(
+                    new RendererCapabilities[] {ALL_AUDIO_FORMAT_SUPPORTED_RENDERER_CAPABILITIES},
+                    singleTrackGroup(lowerBitrateFormat, higherBitrateFormat),
+                    periodId,
+                    TIMELINE);
+
+    assertThat(result.selections.get(0).getSelectedFormat()).isEqualTo(higherBitrateFormat);
```
Comment (tonihei) [inline library/core/src/test/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelectorTest.java:1126]: You should probably add an assertion that only one track is in the selection:
assertThat(result.selections.get(0).length()).isEqualTo(1);

Or to put it the other way round: The test should fail if you comment out the first line (which I think is currently not the case).

File: library/core/src/test/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelectorTest.java [hunk 2]
```diff
@@ -1078,6 +1078,33 @@ public void testSelectTracksWithinCapabilitiesAndForceLowestBitrateSelectLowerBi
     assertThat(result.selections.get(0).getSelectedFormat()).isEqualTo(lowerBitrateFormat);
   }
 
+  /**
+   * Tests that track selector will select audio tracks with higher bitrate when {@link Parameters}
+   * indicate highest bitrate preference, even when tracks are within capabilities.
```
Comment (tonihei) [inline library/core/src/test/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelectorTest.java]: The "even when tracks are within capabilities" part doesn't make much sense here. Maybe say:
"Tests that track selector will select the highest supported audio track when {@link Parameters#forceHighestSupportedBitrate} is set."

File: library/core/src/test/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelectorTest.java [hunk 3]
```diff
@@ -1078,6 +1078,33 @@ public void testSelectTracksWithinCapabilitiesAndForceLowestBitrateSelectLowerBi
...
+   */
+  @Test
+  public void testSelectTracksWithinCapabilitiesAndForceHighestBitrateSelectHigherBitrate()
+          throws Exception {
+    trackSelector.setParameters(new ParametersBuilder().setForceHighestSupportedBitrate(true).build());
+
+    Format lowerBitrateFormat =
+            Format.createAudioSampleFormat("audioFormat", MimeTypes.AUDIO_AAC, null, 15000,
+                    Format.NO_VALUE, 2, 44100, null, null, 0, null);
+    Format higherBitrateFormat =
```
Comment (tonihei) [inline library/core/src/test/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelectorTest.java:1104]: Would be nice to add a third format (with an even higher bitrate) which exceeds the renderer capabilities. See for example testSelectTracksPreferTrackWithinCapabilities for how to configure that.

**Review by tonihei — COMMENTED** [review]
Thanks for the test! Looks good, just some small improvement ideas.

File: library/core/src/test/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelectorTest.java (hunk 1 above)
Comment (BetweenTwoBits) [reply library/core/src/test/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelectorTest.java:1126]: Hmm, should this change also be made for `testSelectTracksWithinCapabilitiesAndForceLowestBitrateSelectLowerBitrate`?

File: library/core/src/test/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelectorTest.java (hunk 2 above)
Comment (BetweenTwoBits) [reply library/core/src/test/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelectorTest.java]: This change also seems applicable to the other test.

File: library/core/src/test/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelectorTest.java (hunk 1 above)
Comment (tonihei) [reply library/core/src/test/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelectorTest.java:1126]: The tracks will be sorted by decending bitrate, so having the lowest bitrate as the first track means there can be only one track. That's very non-obvious though and I agree we should add that to testSelectTracksWithinCapabilitiesAndForceLowestBitrateSelectLowerBitrate as well.

File: library/core/src/test/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelectorTest.java (hunk 2 above)
Comment (tonihei) [reply library/core/src/test/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelectorTest.java]: Yes, the other comment can be improved too. 

Aside: We generally don't provide javadocs for tests as long as the test name itself is descriptive enough. This class seems to be an exception to that rule :)

**Comment by BetweenTwoBits** [comment]
All changes made(in highest and lowest bitrate scenarios). I can't help but notice the naming is slightly different between these methods:
`setForceLowestBitrate`
`setForceHighestSupportedBitrate`

Is there a reason for that difference or could I also throw in: `setForceLowestBitrate` -> `setForceLowestSupportedBitrate`

**Comment by tonihei** [comment]
Nice, thanks! 

The reason for the different naming is that we had `setForceLowestBitrate` only until recently. And in almost all cases the formats at the lower end won't exceed renderer capabilities - so the distinction between just "lowest bitrate" and "lowest supported bitrate" is not really relevant.

When we added  `setForceHighestSupportedBitrate`, we specifically needed to include the "supported" bit because otherwise it would be unclear if it would force the selection to go beyond the renderer capabilities or not. For consistency, I agree that it would be clearer to rename the other parameter as well. But keep in mind that people are already using it and we would break existing implementation unnecessarily.

**Comment by BetweenTwoBits** [comment]
Ah, that all makes sense. Thanks for the review!

**Pull request merged**